src/logging.h

```diff
@@ -167,4 +168,50 @@ static inline void LogPrint(const BCLog::LogFlags& category, const Args&... args
     }
 }
 
+class ILogFilter
+{
+public:
+    virtual ~ILogFilter(){};
+    virtual bool filter() = 0;
+};
+
+// Implementation that logs at most every millis_ milliseconds
+class TimeThrottledFilter : public ILogFilter
+{
+private:
+    uint64_t millis_{0};
+    uint64_t last_time_millis_{0};
```

Comment:
Can we have those 2 variable `atomic_uint64_t` Currently we don't use threads but may change so atomic is more legitimated.

Reply:
Adjusted 👍